dotnet/src/SemanticKernel.Abstractions/AI/IModelResult.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.AI;
+
+/// <summary>
+/// Interface for model results
+/// </summary>
+public interface IModelResult
```

Comment:
nit: I think there may be some confusion between:
- `IModelResult` interface, which is base for `IChatResult`, `ITextResult` etc.
- `ModelResult` class, which is container for results from AI.
- `ModelResult` property, which is property of `IModelResult` interface.

From the first sight, it looks like `ModelResult` class should implement `IModelResult` interface, but it's not the case.

Can we change the name of this interface to `IAIResult` or `IBaseResult` to clearly define that `ModelResult` implementation is unrelated to this interface and it's just part of it (as property).

Reply:
@RogerBarreto Can we visualize the interfaces, the result types their members return, and the classes that implement them as a class diagram in Mermaid or any other tool? The number of interfaces is increasing, and it's becoming quite difficult to follow.